src/features/CanvasEditor.tsx

```diff
@@ -32,144 +43,66 @@ const defaultDefinition = {
   }
 };
 
-const CanvasEditor = () => {
-  // const defaultPageId = uuid();
-  const canvasContainerRef = createRef();
-  const downloadRef = createRef();
-  const selectionRef = useRef();
-  // const [isDragging, setIsDragging] = useState(false);
-  const [appDefinition, setAppDefinition] = useState(defaultDefinition);
-  const [selectedComponents, setSelectedComponents] = useState([]);
-  const [selectedComponent, setSelectedComponent] = useState({});
-  const [hoveredComponent, setHoveredComponent] = useState(null);
-  const [currentPageID, setCurrentPageID] = useState(1);
-  const [zoomLevel, setZoomLevel] = useState(1.0);
-  const [rendering, setRendering] = useState(false);
+/**
+ * CanvasEditor
+ *
+ * Description: This return entire canvas and component library
+ */
 
+const CanvasEditor = () => {
+  const canvasContainerRef = useRef<HTMLInputElement>(null);
+  /**
+   * set isDragging when component Drag
+   */
+  const [isDragging, setIsDragging] = useState(false);
+  /**
+   * set the current Page
+   */
+  const [currentPage, setCurrentPage] = useState<string>(defaultPageId);
+  /**
+   * set the app Definition of the project
+   */
+  const [appDefinition, setAppDefinition] =
+    useState<IAppDefination>(defaultDefinition);
+  /**
+   * set rendering till page gets loaded for first time
+   *  - this is now working  but incase for the first time if
+   *    component is does not drag the we have to add it
+   */
+  // const [rendering, setRendering] = useState(stubTrue);
+
+  const zoomLevel = 1;
+
+  /**
+   * Wait for 500 seconds to page get loaded
+   * - this is now working  but incase for the first time if
+   *    component is does not drag the we have to add it
+   */
   useEffect(() => {
-    const run = debounce(() => {
-      setRendering(true);
-    }, 500);
-    run();
+    // const run = debounce(() => {
```

Comment:
add reason of adding comments, By default we should not keep unnecessary comments

Reply:
Sure